controllers/utils.go

```diff
@@ -0,0 +1,49 @@
+package controllers
+
+import (
+	monitoringv1beta1 "github.com/appuio/scheduler-canary-controller/api/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func buildPodFromTemplate(template *corev1.PodTemplateSpec, instance *monitoringv1beta1.SchedulerCanary) (*corev1.Pod, error) {
+	labels := copyStringMap(template.Labels)
+	labels[instanceLabel] = instance.Name
+
+	annotations := copyStringMap(template.Annotations)
+	annotations[timeoutAnnotation] = instance.Spec.MaxPodCompletionTimeout.Duration.String()
```

Comment:
Template deep copt would give the wrong type. But maybe meta also has a deep copy option.

Reply:
Can't you do this?
```
func buildPodFromTemplate(template *corev1.PodTemplateSpec, instance *monitoringv1beta1.SchedulerCanary) (*corev1.Pod, error) {
	tempCopy := template.DeepCopy()
	pod := &corev1.Pod{
		ObjectMeta: metav1.ObjectMeta{
			Namespace:    instance.GetNamespace(),
			Labels:       tempCopy.Labels,
			Annotations:  tempCopy.Annotations,
			GenerateName: suffixLimit(instance.GetName(), "-"),
			Finalizers:   copyStringSlice(template.Finalizers),
		},
		Spec: tempCopy.Spec,
	}
	return pod, nil
}
```

Reply:
Yep that's what i thought too. Makes the code not cleaner though since deep copy preserves the nil maps. and i need to initialise them later.

I found another way though to remove the ugly copy helpers

Reply:
Ah right nil checks..  Fair enough, using `x/exp` libraries works too (just something we'll eventually need to update, but that shouldn't be too big of an issue)